src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Models/ProvideSemanticTokensResponse.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+#pragma warning disable CS0618
+using System;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models.Proposals;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Semantic
+{
+    public class ProvideSemanticTokensResponse
+    {
+        public SemanticTokens Result { get; }
+
+        public long? HostDocumentSyncVersion { get; }
+
+        public ProvideSemanticTokensResponse(SemanticTokens result, long? hostDocumentSyncVersion)
+        {
+            Result = result;
+            HostDocumentSyncVersion = hostDocumentSyncVersion;
+        }
+
+        public override bool Equals(object obj)
+        {
+            if(obj is ProvideSemanticTokensResponse)
+            {
+                var other = obj as ProvideSemanticTokensResponse;
+
+                return other.HostDocumentSyncVersion.Equals(HostDocumentSyncVersion) && other.Result.Equals(Result);
```

Comment:
Hmm do we actually utilize this equals anywhere? Only asking because technically the same host document sync version can result in different "provide" results due to another C# file changing. Wondering if we should just remove the equals entirely

Reply:
This Equals is only used in our tests to ensure we created something equivalent to the expected Response. If our current SDK supported it I would have just made it a `record` for that sweet sweet C#9 goodness.